flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1277,7 +1277,7 @@ public void requestPartitionProducerState(
      * @param checkpointTimestamp The timestamp associated with the checkpoint.
      * @param checkpointOptions Options for performing this checkpoint.
      */
-    public void triggerCheckpointBarrier(
+    public CompletableFuture<Boolean> triggerCheckpointBarrier(
```

Comment:
nit: if we do need to return the result, can we return a `Future` (not a `CompletableFuture`), which can't be completed outside accidentially?